test/ng/directive/asSpec.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+describe('as', function() {
+
+  it('should bind in the current scope the controller of a component', function() {
+    var expected = {some: 'value'};
+
+    angular.module('my', []).component('myComponent', {
+      template: 'foo',
+      controller: function() {
+        this.property = expected;
+      }
+    });
+    module('my');
+
+    inject(function($compile, $rootScope) {
+      var scope = $rootScope.$new();
+      var $ctrl = {undamaged: true};
+      scope.$ctrl = $ctrl;
+      dealoc($compile('<my-component as="$ctrl.myComponent"></my-component>')(scope));
+
+      expect(scope.$ctrl).toBe($ctrl);
+      expect(scope.$ctrl.undamaged).toBe(true);
+      expect($ctrl.myComponent).not.toBeUndefined();
+      expect($ctrl.myComponent && $ctrl.myComponent.property).toBe(expected);
+      scope.$destroy();
+    });
+  });
+
+  it('should be parametrized with any variable', function() {
+    var myComponentController;
+
+    angular.module('my', []).component('myComponent', {
+      template: 'foo',
+      controller: function() {
+        myComponentController = this;
+      }
+    });
+    module('my');
+
+    inject(function($compile, $rootScope) {
+      var scope = $rootScope.$new();
+      dealoc($compile('<my-component as="bar.myComponent"></my-component>')(scope));
+
+      expect(scope.bar).not.toBeUndefined();
+      expect(scope.bar && scope.bar.myComponent).toBe(myComponentController);
+      scope.$destroy();
+    });
+  });
+
+  it('should be compatible with entity directives with controller', function() {
+    var myDirectiveController;
+
+    angular.module('my', []).directive('myDirective', function() {
+      return {
+        restrict: 'E',
+        controller: function() {
+          myDirectiveController = this;
+        }
+      };
+    });
+    module('my');
+
+    inject(function($compile, $rootScope) {
+      var scope = $rootScope.$new();
+      dealoc($compile('<my-directive as="bar.myDirective"></my-directive>')(scope));
+
+      expect(scope.bar).not.toBeUndefined();
+      expect(scope.bar && scope.bar.myDirective).toBe(myDirectiveController);
+      scope.$destroy();
+    });
+  });
+
+  it('should bind non HTMLElements elements into the current scope if no component', inject(function($compile, $rootScope) {
```

Comment:
You are right, I missed that. BTW, you don't need to create a new module with `angular.module(...)`. (The `.component()` tests do it for different reasons.)

You might be able to simplify the tests by creating the component in a `beforeEach()` block, but it's unimportant at this point. It's fine as it is.